### PR TITLE
fix(ext/node): add `truncate` method to the `FileHandle` class

### PR DESCRIPTION
**Pull request by filipstev**
- Implements `FileHandle.truncate`
- Adds a promisified `fs.ftruncate` under `fs/promises` which is used internally by `FileHandle.ftruncate`
- Adds a couple of formatting changes due to running `./tools/format.js`
- Adds tests
- `filehandle.truncate` Node documentation: [click here](https://nodejs.org/api/fs.html#filehandletruncatelen)

part of https://github.com/denoland/deno/issues/25554

### CODE REVIEW

**Comment by filipstev** [comment]
Hey @kt3k, sorry for pinging you, could you take a look at this PR?
Happy to keep working on it if there are any further suggestions!

File: ext/node/polyfills/fs.ts [hunk 1]
```diff
@@ -165,6 +169,7 @@ const promises = {
   opendir: opendirPromise,
   rename: renamePromise,
   truncate: truncatePromise,
+  ftruncate: ftruncatePromise,
```
Comment (kt3k) [inline ext/node/polyfills/fs.ts:172]: Node doesn't seem having `fs.promises.ftruncate` (No doc entry for it, and also not available in runtime) https://nodejs.org/api/fs.html

Let's not expose it

File: ext/node/polyfills/fs/promises.ts [hunk 2]
```diff
@@ -8,6 +8,7 @@ export const open = fsPromises.open;
 export const opendir = fsPromises.opendir;
 export const rename = fsPromises.rename;
 export const truncate = fsPromises.truncate;
+export const ftruncate = fsPromises.ftruncate;
```
Comment (kt3k) [inline ext/node/polyfills/fs/promises.ts:11]: Ditto. Let's not expose this

**Comment by kt3k** [comment]
Thanks for the PR!

The test cases look good. I left some comments.

File: ext/node/polyfills/fs.ts (hunk 1 above)
Comment (filipstev) [reply ext/node/polyfills/fs.ts:172]: You make a great point! I've noticed that `fstat` is also exposed but can't seem to find it in the doc, and `fs.promises.fstat` also doesn't seem to be available in the runtime. Is that an issue, and if so, is it something we would like to handle within this PR?

Comment (kt3k) [reply ext/node/polyfills/fs.ts:172]: > fstat is also exposed but can't seem to find it in the doc, and fs.promises.fstat also doesn't seem to be available in the runtime. Is that an issue

Good point. Yes, that seems an issue to me.

>, and if so, is it something we would like to handle within this PR?

Fixing it in this PR is welcome, but that's not absolutely necessary in my opinon. I created an issue for `fs.promises.fstat` for now https://github.com/denoland/deno/issues/27423

Comment (filipstev) [reply ext/node/polyfills/fs.ts]: Awesome, I might take a deeper look into that one as well :)

Also, the implementation for `promises.ftruncate` has been cleaned up 🧹

**Review by kt3k — APPROVED** [review]
LGTM. Nice work! Thanks for your contribution

**Pull request merged**